### ISSUE TITLE
service istiod not found error at "kfctl 1.0.2 with kfctl_k8s_istio.v1.0.2.yaml "

### ISSUE THREAD

**Issue by insoo67park** [kubeflow/kubeflow#5075]
/kind bug

1. My private k8s cluster (version: 1.17.4)
2. I tried to install kubeflow and got error message as below
   Encountered error applying istio-install: (kubeflow error): Code 500 with message: Apply run Error error when creating "/tmp/kout701552754": Internal error occured: failed calling webhook "validation.istio.io": "Post: https://istiod.istio-system.svc:443/validate?timeout=30s: service "istiod" not found filename="kustomize/kustomize.go:202"
Will retry in 8 seconds. filename="kustomize/kustomize.go:203"

**What did you expect to happen:**
kubeflow 1.0.2 should be installed successfully in my private k8s cluster.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
I found that kubeflow 1.0.2 uses istio version 1.1.6. And also I found **istiod** is new daemon service from istio version 1.5 or later.
I check if istiod service created in my k8s cluster. No istiod service is not created forever.

**Environment:**
- Kubeflow version: (version number can be found at the bottom left corner of the Kubeflow dashboard):
- kfctl version: 1.0.2
- Kubernetes platform: (kubectl, 1.17)
- Kubernetes version: 1.17.4:
- OS (centos 7):

**Etc**
I don't know why kubeflow version 1.0.2 try to connect isdtiod service which is not created forever.
Is this bug or my mistake?
Does anyone know this issue and has solution?

Thanks in advance.

**Comment by 8bitmp3** [comment]
Hi @insoo67park, can you give us a step-by-step flow of how you installed KF 1.0.2? There're various deployment options on https://www.kubeflow.org/docs/started/getting-started/, so it'd be good if you specify how you approached it, so we can help.

> istiod is new daemon service from istio version 1.5 

The `istiod` control plane was indeed introduced from Istio v1.5 encapsulating Pilot, Citadel, etc.

**Comment by insoo67park** [comment]
Hi @8bitmp3

In fact, I installed istio v1.5 into my private kubernetes cluster (v1.17.4) before trying install kubeflow v1.0.2  (istio with dex version).

My steps are below
1. I deleted istio-system namespace from my kubernetes cluster (kubectl delete namespace istio-system) - I thought this would delete all resources related istio v1.5
   -> I checked if all the services/pods/deploys are deleted from kubernetes cluster)
2. I downloaded kfctl from https://github.com/kubeflow/kfctl/releases/
3. I executed command (tar -xvf kfctl_v1.0.2-0-ga476281_linux.tar.gz)
4. I executed command flow as said at https://www.kubeflow.org/docs/started/k8s/kfctl-istio-dex/
   -> export PATH=$PATH:/home/download/kubeflow
   -> export CONFIG_URI="https://raw.githubusercontent.com/kubeflow/manifests/v1.0-branch/kfdef/kfctl_istio_dex.v1.0.2.yaml"
   -> export KF_NAME=kf-test
   -> export BASE_DIR=/home/download/kubeflow
   -> export KF_DIR=${BASE_DIR}/${KF_NAME}
   -> mkdir -p ${KF_DIR}, cd ${KF_DIR}
   -> wget -O kfctl_istio_dex.yaml $CONFIG_URI
   -> export CONFIG_FILE=${KF_DIR}/kfctl_istio_dex.yaml
   -> kfctl build -V -f ${CONFIG_FILE}
   -> kfctl apply -V -f ${CONFIG_FILE}
5. Something normal logs are passed on my console, and after that, a message trying to connect to  "https://istiod.istio-system.svc:443/validate"  and fail to find istiod service in k8s was appeared on my console repeatably.

In fact, I tried to install in GCP's VM  with microk8s environment. But in GCP's VM environment, I also met a error message related to istio's valiation (but not exact same with above message)
Only case I success to intall kubeflow is below case.
1. GCP VM enviromnent
2, microk8s
3. in kicrok8s, I execute a command (microk8s enable kubeflow)

I think, install for kubeflow is not esay with istio and it is specially informed to users how to install KF with istio.

Thanks a lot in advance.

**Comment by insoo67park** [comment]
I found that the same error message is in "istio-security-post-install-release-1.3-latest-daily-xxx" pod. (kubectl logs istio-security-post-install-release-1.3-latest-daily-xxx -n istio-system)

I guess, something check process is done in istio-security-post-install pod and at there it refers to istiod service which is not exist.

I hopes this information would be helpful for your analysis.

**Comment by yanniszark** [comment]
This is weird and should not happen. Maybe upstream Istio pushed an image that doesn't work with current 1.3 installations? I would open an issue in upstream Istio (if none exists already), provide the image that produced the error and ask about it.

**Comment by insoo67park** [comment]
@yanniszark 
Could you share the issue progressing status for me?

Thanks a lot in advance.

**Comment by EduardFazliev** [comment]
I have the same issue not only using version 1.0.2, I've tried 0.7 version and had same error... Is there a workaround for this?

**Comment by insoo67park** [comment]
Thanks for your help.

I solved this problem with upgrading to kubeflow version v1.1.0 (and from my machine to vm on google cloud platform)
But I think, in my case root cause is my networking environment. 
I guess some docker image or other things did not communicated as needed.

I close this issue.

**Comment by ccll** [comment]
Same problem here trying to install kubeflow v1.1.0 in on-prem k8s.

```
$ kfctl apply -V -f kfctl_k8s_istio.v1.1.0.yaml
kfctl exited with error: failed to apply:  (kubeflow.error): Code 500 with message: kfApp Apply failed for kustomize:  (kubeflow.error): Code 500 with message: Apply.Run : error when creating "/tmp/kout494903682": Internal error occurred: failed calling webhook "validation.istio.io": Post "https://istiod.istio-system.svc:443/validate?timeout=30s": service "istiod" not found
```

The 'istiod' service is never created.

```
$ kubectl get svc -n istio-system
NAME                     TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                                                                                                                                      AGE
istio-citadel            ClusterIP   10.43.68.7      <none>        8060/TCP,15014/TCP                                                                                                                           23h
istio-galley             ClusterIP   10.43.237.49    <none>        443/TCP,15014/TCP,9901/TCP                                                                                                                   23h
istio-ingressgateway     NodePort    10.43.160.63    <none>        15020:32170/TCP,80:31380/TCP,443:31390/TCP,31400:31400/TCP,15029:30093/TCP,15030:31454/TCP,15031:30400/TCP,15032:31758/TCP,15443:32747/TCP   23h
istio-pilot              ClusterIP   10.43.103.208   <none>        15010/TCP,15011/TCP,8080/TCP,15014/TCP                                                                                                       23h
istio-policy             ClusterIP   10.43.68.242    <none>        9091/TCP,15004/TCP,15014/TCP                                                                                                                 23h
istio-sidecar-injector   ClusterIP   10.43.238.82    <none>        443/TCP,15014/TCP                                                                                                                            23h
istio-telemetry          ClusterIP   10.43.85.115    <none>        9091/TCP,15004/TCP,15014/TCP,42422/TCP                                                                                                       23h
prometheus               ClusterIP   10.43.200.89    <none>        9090/TCP
```

**Environment**
Clean rancher k3s cluster, v1.19.4, single master node, no workers.
kfctl: `kfctl v1.1.0-0-g9a3621e`

**Comment by summerisc** [comment]
Same here trying to install kubeflow v1.2.0 in on-prem k8s. Any solutions/causes to this?

**Comment by huboqiang** [comment]
Hi,  I met the same problem, and I solved it by removing existed istio v1.7 by ` istioctl x uninstall --purge` . 

You can try to  unisntall istio v1.7 and re-apply the yaml file.
